attestation-agent/kbs_protocol/src/client/rcar_client.rs

```diff
@@ -309,10 +309,21 @@ impl KbsClient<Box<dyn EvidenceProvider>> {
 #[async_trait]
 impl KbsClientCapabilities for KbsClient<Box<dyn EvidenceProvider>> {
     async fn get_resource(&mut self, resource_uri: ResourceUri) -> Result<Vec<u8>> {
-        let mut remote_url = format!(
-            "{}/{KBS_PREFIX}/resource/{}/{}/{}",
-            self.kbs_host_url, resource_uri.repository, resource_uri.r#type, resource_uri.tag
-        );
+        let mut remote_url = if resource_uri.repository == "plugin" {
+            // plugin case → format: {host}/{KBS_PREFIX}/{plugin_name}/{tag}
+            format!(
+                "{}/{KBS_PREFIX}/{}/{}",
+                self.kbs_host_url,
+                resource_uri.r#type, // plugin name
+                resource_uri.tag
+            )
+        } else {
+            // resource plugin case → format: {host}/{KBS_PREFIX}/resource/{repository}/{type}/{tag}
+            format!(
+                "{}/{KBS_PREFIX}/resource/{}/{}/{}",
+                self.kbs_host_url, resource_uri.repository, resource_uri.r#type, resource_uri.tag
```

Comment:
Note that we also have [query](https://github.com/confidential-containers/guest-components/blob/main/attestation-agent/deps/resource_uri/src/lib.rs#L25) field here

Reply:
Oh yes. L327-L329 I just missed 😓